### PR TITLE
ReadMap, panic-proof

### PR DESCRIPTION
**Pull request by anthonybishopric**
Two more features commonly used to read free-form YAML config.

### CODE REVIEW

**Comment by tummychow** [comment]
:+1:

**Comment by petertseng** [comment]
ok but it looks like you need to fix config_test.go

File: pkg/config/config.go [hunk 1]
```diff
@@ -32,14 +33,41 @@ func LoadConfigFile(filepath string) (*Config, error) {
...
 }
 
 func (c *Config) Read(key string) interface{} {
 	return c.unpacked[key]
 }
+
+func (c *Config) ReadMap(key string) (*Config, error) {
+	readVal := c.Read(key)
+	if readVal == nil {
+		return &Config{make(map[interface{}]interface{})}, nil
```
Comment (petertseng) [inline pkg/config/config.go]: whoa

**Comment by anthonybishopric** [comment]
Tentatively going to leave the golang downgrade in here. If the untar error continues i'll upgrade to 1.4 again.

**Pull request merged**